pkg/diagnostics/analyze.go

```diff
@@ -0,0 +1,319 @@
+package diagnostics
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/backingstore"
+	"github.com/noobaa/noobaa-operator/v5/pkg/bundle"
+	"github.com/noobaa/noobaa-operator/v5/pkg/namespacestore"
+	"github.com/noobaa/noobaa-operator/v5/pkg/options"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/spf13/cobra"
+)
+
+// RunAnalyzeBackingStore runs a CLI command
+func RunAnalyzeBackingStore(cmd *cobra.Command, args []string) {
+	backingStore := backingstore.GetBackingStoreFromArgs(cmd, args)
+	util.Logger().Printf("⏳ Running this command will run tests on BackingStore %s...\n", backingStore.Name)
+	collector := collectorInstance(cmd)
+	makeDirForLogs(collector)
+	analyzeBackingstore(cmd, backingStore, collector)
+	destDir, _ := cmd.Flags().GetString("dir")
+	collector.ExportDiagnostics(destDir)
+	collector.log.Println("✅ Logs were saved")
+}
+
+// RunAnalyzeNamespaceStore runs a CLI command
+func RunAnalyzeNamespaceStore(cmd *cobra.Command, args []string) {
+	namespaceStore := namespacestore.GetNamespaceStoreFromArgs(cmd, args)
+	util.Logger().Printf("⏳ Running this command will run tests on NamespaceStore %s...\n", namespaceStore.Name)
+	collector := collectorInstance(cmd)
+	makeDirForLogs(collector)
+	analyzeNamespaceStore(cmd, namespaceStore, collector)
+	destDir, _ := cmd.Flags().GetString("dir")
+	collector.ExportDiagnostics(destDir)
+	collector.log.Println("✅ Logs were saved")
+}
+
+// RunAnalyzeResources runs a CLI command
+func RunAnalyzeResources(cmd *cobra.Command, args []string) {
+	collector := collectorInstance(cmd)
+	makeDirForLogs(collector)
+	collector.log.Println("Running this command will run tests on all your resources")
+	collector.log.Println("Iterating over all BackingStores...")
+	analyzeAllBackingstores(cmd, collector)
+	collector.log.Println("Iterating over all NamespaceStore...")
+	analyzeAllNamespacestores(cmd, collector)
+	destDir, _ := cmd.Flags().GetString("dir")
+	collector.ExportDiagnostics(destDir)
+	collector.log.Println("✅ Logs were saved")
+}
+
+func analyzeAllBackingstores(cmd *cobra.Command, collector *Collector) {
+	list := &nbv1.BackingStoreList{
+		TypeMeta: metav1.TypeMeta{Kind: "BackingStoreList"},
+	}
+	if !util.KubeList(list, &client.ListOptions{Namespace: options.Namespace}) {
+		collector.log.Printf("Could not get BackingStoreList.\n")
+		return
+	}
+	if len(list.Items) == 0 {
```

Comment:
I suggest filtering out all irrelevant backing stores (pv-pools, sts, etc.). otherwise, we might create an empty zip file if there are only pv-pools